### ISSUE TITLE
IgxGrid: form control error when adding dynamic columns

### ISSUE THREAD

**Issue by ddaribo** [IgniteUI/igniteui-angular#13025]
## Description  
For a grid, in which columns can be dynamically added at run-time, form controls for the newly added fields' cells are missing. This is valid for cells in records that were already edited, so their `formGroup` was created prior to adding the new columns. The result is error on attempting to edit these cells.

 * igniteui-angular version: 14.2.x, 15.1.x, 16.0.x 
 * browser: any

## Steps to reproduce  

1. Open [this sample](https://stackblitz.com/edit/igx-grid-dynamic-columns-add-column?file=src/app/grid/grid-column-data-types-sample/grid-column-data-types-sample.component.ts).
2. Click the "Add Column" button.
3. Double click on the "Column5" cell in the first record to edit it.
4. Leave edit mode.
5. Click the "Add Column" button again.
6. Double click on the "Column6" cell **again in the first row** to edit it -> View the error in the console.
7. Double click on the "Column6" cell **in the second row** to edit it -> No new error in the console.

## Result  
On Step 6:
```
ERROR Error: Cannot find control with unspecified name attribute
    at _throwError (vendor.js:66251:9)
    at setUpControl (vendor.js:66042:19)
    at FormControlDirective.ngOnChanges (vendor.js:67829:7)
    at ...
```

## Expected result  
Being able to edit the cells without errors.

## Attachments  
The first row's cell editor input does not have a `ng-reflect-form` attribute or the other validation attributes.
![image](https://github.com/IgniteUI/igniteui-angular/assets/35433937/85026ac5-bd66-4455-b692-7b711069f604)

The second row's cell editor input has a `ng-reflect-form` attribute as well as others.
![image](https://github.com/IgniteUI/igniteui-angular/assets/35433937/e5176f00-1494-4e01-bbe7-7e059c3d0ef0)

**Comment by Tuna93z** [comment]
Hi @ddaribo !
Has the above error been fixed yet?

**Comment by ddaribo** [comment]
Hi @Tuna93z, 
The pending PRs are yet to be verified. Once they are, the fix will be available in one of the upcoming releases of 14.2.x, as there is a PR for this branch as well.

**Comment by Tuna93z** [comment]
Hi @ddaribo,
This issue has been closed. Is the task fixed yet? 
Why do I still have the error when I run it again?

**Comment by ddaribo** [comment]
Hi @Tuna93z,
As per our [Releases page](https://github.com/IgniteUI/igniteui-angular/releases), the fix has entered the 14.2.3 release. Please, make sure to update to it.

For quick reference, [here](https://stackblitz.com/edit/igx-grid-dynamic-columns-add-column?file=package.json) is the test sample with the release (against version 15) containing the fix. The error does not seem to be reproduced after updating.

**Comment by Tuna93z** [comment]
Hi @ddaribo,
I tested version 14.2.33 and the error was fixed
Tks you!
